pkg/operator/collection.go

```diff
@@ -104,6 +109,11 @@ func setupCollectionControllers(op *Operator) error {
 				objFilterCollector,
 				predicate.GenerationChangedPredicate{},
 			)).
+		// Detect and undo changes to the secret.
+		Watches(
+			source.NewKindWithCache(&corev1.Secret{}, op.managedNamespacesCache),
```

Comment:
Is using a dedicated, separate informer cache necessary when you've defined the constrained one already [here](https://github.com/GoogleCloudPlatform/prometheus-engine/blob/8e89c873d3bc21f735941b968461c5e48c153d9f/pkg/operator/operator.go#L177)? I'm assuming that's the cache that's injected by default.